quinn-udp/src/unix.rs

```diff
@@ -444,16 +483,26 @@ fn prepare_msg(
                     };
                     encoder.push(libc::IPPROTO_IP, libc::IP_PKTINFO, pktinfo);
                 }
-                IpAddr::V6(v6) => {
-                    let pktinfo = libc::in6_pktinfo {
-                        ipi6_ifindex: 0,
-                        ipi6_addr: libc::in6_addr {
-                            s6_addr: v6.octets(),
-                        },
-                    };
-                    encoder.push(libc::IPPROTO_IPV6, libc::IPV6_PKTINFO, pktinfo);
+                #[cfg(any(target_os = "freebsd", target_os = "macos"))]
+                {
+                    if encode_srcip {
```

Comment:
It looks like the macOS version of `send` always passes `false` for `encode_srcip`, rendering this code dead on macOS, even though the cfg gate ensures it's compiled. Is that intended?

Reply:
Sorry, corrected now.